libpod/kube.go

```diff
@@ -0,0 +1,277 @@
+	result, err := ContainerToV1Container(ctr)
+	if err != nil {
+		return nil, err
+	}
+	containers = append(containers, result)
+
+	tm := v12.TypeMeta{
+		Kind:       "Pod",
+		APIVersion: "v1",
+	}
+
+	// Add a label called "app" with the containers name as a value
+	labels := make(map[string]string)
+	labels["app"] = removeUnderscores(ctr.Name())
+	om := v12.ObjectMeta{
+		// The name of the pod is container_name-libpod
+		Name:   fmt.Sprintf("%s-libpod", removeUnderscores(ctr.Name())),
+		Labels: labels,
+		// Seems required so added this, else shows null
+		CreationTimestamp: v12.Now(),
+	}
+	ps := v1.PodSpec{
+		Containers: containers,
+	}
+	p := v1.Pod{
+		TypeMeta:   tm,
+		ObjectMeta: om,
+		Spec:       ps,
+	}
+	return &p, nil
+}
+
+// ContainerToV1Container converts information we know about a libpod container
+// to a V1.Container specification.
+func ContainerToV1Container(c *Container) (v1.Container, error) {
+	container := v1.Container{}
+	kubeSec, err := generateKubeSecurityContext(c)
+	if err != nil {
+		return container, err
+	}
+
+	if len(c.Spec().Linux.Devices) > 0 {
+		// TODO Enable when we can support devices and their names
+		devices, err := generateKubeVolumeDeviceFromLinuxDevice(c.Spec().Linux.Devices)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeDevices = devices
+		return container, errors.Wrapf(ErrKubeNotSupported, "linux devices")
+	}
+
+	if len(c.config.UserVolumes) > 0 {
+		// TODO When we until we can resolve what the volume name should be, this is disabled
+		volumes, err := libpodMountsToKubeVolumeMounts(c)
+		if err != nil {
+			return container, err
+		}
+		container.VolumeMounts = volumes
+		return container, errors.Wrapf(ErrKubeNotSupported, "volume names")
+	}
+
+	containerCommands := c.Command()
+	container.Name = removeUnderscores(c.Name())
+
+	_, image := c.Image()
+	container.Image = image
+	container.Stdin = c.Stdin()
+	container.Command = containerCommands
+	// TODO need to figure out how we handle command vs entry point.  Kune appears to prefer entrypoint.
+	// right now we just take the container's command
+	//container.Args = args
+	container.WorkingDir = c.WorkingDir()
+	container.Ports = ocicniPortMappingToContainerPort(c.PortMappings())
+	// This should not be applicable
+	//container.EnvFromSource =
+	container.Env = libpodEnvVarsToKubeEnvVars(c.config.Spec.Process.Env)
+	// TODO enable resources when we can support naming conventions
+	//container.Resources
+	container.SecurityContext = kubeSec
+	container.Stdin = c.Stdin()
+	container.StdinOnce = false
+	container.TTY = c.config.Spec.Process.Terminal
+
+	return container, nil
+}
+
+// ocicniPortMappingToContainerPort takes an ocicni portmapping and converts
+// it to a v1.ContainerPort format for kube output
+func ocicniPortMappingToContainerPort(portMappings []ocicni.PortMapping) []v1.ContainerPort {
+	var containerPorts []v1.ContainerPort
+	for _, p := range portMappings {
+
+		protocol := v1.ProtocolTCP
```

Comment:
Maybe do a switch here instead, to catch invalid protocols

Reply:
uh, we cant have an invalid protocol; this came from libpod.  Still want it?

Reply:
agreed to do